File: ifif/python/S2_19583.py
# 문자열
# 싸이버개강총회

import sys
sys.stdin = open('input.txt', 'r')
input = sys.stdin.readline
S, E, Q = input().split()
S = list(map(int, S.split(":")))
E = list(map(int, E.split(":")))
Q = list(map(int, Q.split(":")))

name_dict = dict()
cnt = 0
def checkStart(time):
    if S[0]> time[0]:
        return True
    if S[0] == time[0] and S[1] >= time[1]:
        return True
    return False

def checkEnd(time):
    if E[0] < time[0]:
        return True
    if E[0] == time[0] and E[1] <= time[1]:
        return True
    return False

def checkQuit(time):
    if Q[0] > time[0]:
        return True
    if Q[0] == time[0] and Q[1] >= time[1]:
        return True
    return False

while True:
    chat = input().split()
    try:
        time = list(map(int, chat[0].split(":")))
    except:
        break
    person = chat[1]
    if checkStart(time):
        name_dict[person] = 0
    elif checkEnd(time) and checkQuit(time):
        num = name_dict.get(person)
        if num==0:
            name_dict[person] = 1
for v in name_dict.values():
    if v:
        cnt += 1
print(cnt)

File: ifif/java/S2_19583.java
// 문자열
// 싸이버개강총회

import java.io.*;
import java.util.*;

public class S2_19583 {
    public static void main(String[] args) throws IOException {
        System.setIn(new FileInputStream("input.txt"));
        BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
        String[] line = sc.readLine().split(" ");
        int S = convertNumber(line[0]);
        int E = convertNumber(line[1]);
        int Q = convertNumber(line[2]);

        Map<String, Integer> nameMap = new HashMap<>();
        int cnt = 0;
        String[] chat;
        while (true) {
            try {
                chat = sc.readLine().split(" ");
            }
            catch(Exception e) {
                break;
            }
            int curr = convertNumber(chat[0]);
            if (curr <= S) {
                nameMap.put(chat[1], 0);
            } else if (E <= curr && curr <= Q) {
                if (nameMap.getOrDefault(chat[1], -1) == 0) {
                    nameMap.replace(chat[1], 1);
                }
            }
        }
        for (int i : nameMap.values()) {
            if (i==1) {
                cnt += 1;
            }

        }
        System.out.println(cnt);
    }
    private static int convertNumber(String time) {
        String[] timeArray = time.split(":");
        return Integer.parseInt(timeArray[0])*60 + Integer.parseInt(timeArray[1]);
    }
}
